### ISSUE TITLE
Support for Wireless Rival 650?

### ISSUE THREAD

**Issue by Refzlund** [flozz/rivalcfg#112]
I have my wireless mouse (Steelseries Rival 650) connected, but I cannot set any options for it in the terminal.

I simply installed the Archlinux AUR package **rivalcfg-git**.

**Comment by flozz** [comment]
Hello,

This mouse is not supported by rivalcfg yet, sorry. Maybe you can help us to support it? :)

I started to write a doc about contributing to this project: https://github.com/flozz/rivalcfg/blob/rivalcfg4.0/doc/contributing.rst#unsupported-devices

**Comment by urcadox** [comment]
Hello,

I would gladly help to add support for this mouse.

First off, here is the output of `lsusb -d 1038:`:

```
Bus 001 Device 062: ID 1038:1726 SteelSeries ApS USB2.0 Hub
Bus 001 Device 063: ID 1038:172b SteelSeries ApS
```

 - `1038:1726` is the wireless dongle
 - `1038:172b` is the plugged-in mouse

Store page: https://steelseries.com/gaming-mice/rival-650-wireless

Now, how do you reverse engineer this? Trial and error? A tool to capture what's happening on Windows?

**Comment by flozz** [comment]
Hello,

> Now, how do you reverse engineer this? Trial and error? A tool to capture what's happening on Windows?

You are right, there is a tool to capture what's happening on Windows :)

I wrote an article (in French) about the reverse engineering of the Rival 100:

* https://blog.flozz.fr/2016/03/27/steelseries-rival-100-reverse-engineering-dun-peripherique-usb/

I also wrote an other article about the Artcis 7 that can help too:

* https://blog.flozz.fr/2020/05/25/reverse-engineering-recuperer-le-niveau-de-batterie-du-casque-sans-fil-steelseries-arctis-7/

----

In the case of the Rival 650, check that it works the same with the dongle and when the mouse is directly plugged to the computer.

I encourage you to compare what you will find with the [Rival 600](https://github.com/flozz/rivalcfg/blob/rivalcfg4.0/rivalcfg/devices/rival600.py), maybe it work the same, but with some additional options related to the battery. :)

**Comment by urcadox** [comment]
> I wrote an article (in French) about the reverse engineering of the Rival 100:
>
>  - https://blog.flozz.fr/2016/03/27/steelseries-rival-100-reverse-engineering-dun-peripherique-usb/

Great, I'm French so it's not an issue. :)

I will read your article and try this out (maybe today, maybe in 2 months, we'll see).

**Comment by flozz** [comment]
Yeah, do it when you want, and do not hesitate to ask if you need help. I am available here and on Gitter ([![Gitter](https://badges.gitter.im/gitter.svg)](https://gitter.im/rivalcfg/Lobby)). :)

**Comment by dododome01** [comment]
Hello, ive just bought the 650 and im willing to help implement it.
If i do all the reverse engineering and provide you with the data, will you add it to the program or do i have to do it myself?

**Comment by flozz** [comment]
Hello,

SteelSeries send me a Rival 650, so I will work on it. I have to finish with the Sensei TEN and the Rival 500 before, and then I will work the the Rival 650 :)

The devices I am working on:

![](https://blog.flozz.fr/2020/08/24/rivalcfg-comment-jai-cree-un-peu-par-hasard-cet-outil-de-configuration-pour-les-souris-steelseries/images/steelseries_device_sample.jpg)

**Comment by dododome01** [comment]
Seems like a lot of work;
If you need help with the 650 let me know, im just not a big help with coding stuff, since i have no Linux/Python experience!

**Comment by flozz** [comment]
Yes, if I need help I will tell you (I will need help for testing when the device will be implemented).

Currently the progress of the work on the other devices is:

* Sensei TEN: reverse engineering almost finished, I have to start the implementation. It take some time because I started to do it on live on Twitch, so whant to finish its implementation on live too.
* Rival 500: already partially supported, I have to work on button mapping, and on LED colors (but this last feature can wait a future version)

For the Rival 650, I hope it works almost the same as the Rival 600 that is already supported so its initial support can be quick to provide. :)

**Comment by Calosis** [comment]
I recently just bought myself the Rival 650, so i'm happy to help test. I also have the Rival 500, can help test using that as well.

**Comment by flozz** [comment]
@Calosis I have almost finished the work on the Rival 500 (I think it will be done in 1 or 2 live on Twitch, so it should be finished by the end of the next week). Then I will start working on the Rival 650. :)

**Comment by HolyStephano** [comment]
I just got a 650 today, I would be more than willing to do some captures however I can. I have a Windows 10 install on it with steelseries engine as well.

**Comment by flozz** [comment]
@HolyStephano I have this mouse and I already started working on it. I have some notes on a notebook but I dot not started to write code yet :)

**Comment by flozz** [comment]
Yes I know I should work again on this mouse ^^'

**Comment by AreoByte** [comment]
:pray: :pleading_face:

**Comment by dvtkrlbs** [comment]
Are there any progress on this? Mostly for the rgb support. Willing to help both on the reverse engineering and implementation side. Thanks

**Comment by flozz** [comment]
I must admit I have not much time to work on this currently, so yeah... some help will be appreciated ^^'

**Comment by Rouble** [comment]
I've been working on reverse engineering the USB data for setting colors today, figured I'd share what I've got so far
![image](https://github.com/user-attachments/assets/fb275a49-8d06-452f-815c-dd768ec255c7)
